// react-csv-viewer/dist/CsvInterface.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireWildcard(require("react"));

var _papaparse = _interopRequireDefault(require("papaparse"));

require("react-table/react-table.css");

var _reactTable2 = _interopRequireDefault(require("react-table"));

var _CsvInput = _interopRequireDefault(require("./CsvInput"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _nonIterableRest(); }

function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance"); }

function _iterableToArrayLimit(arr, i) { var _arr = []; var _n = true; var _d = false; var _e = undefined; try { for (var _i = arr[Symbol.iterator](), _s; !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }

function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }

function Interface() {
  var _useState = (0, _react.useState)([]),
      _useState2 = _slicedToArray(_useState, 2),
      data = _useState2[0],
      setData = _useState2[1];

  var _useState3 = (0, _react.useState)([]),
      _useState4 = _slicedToArray(_useState3, 2),
      columns = _useState4[0],
      setColumns = _useState4[1];

  var _useState5 = (0, _react.useState)(true),
      _useState6 = _slicedToArray(_useState5, 2),
      loading = _useState6[0],
      setLoading = _useState6[1];

  (0, _react.useEffect)(function () {
    if (data.length && columns.length) setLoading(false);
  }, [data, columns]);

  var handleFileChange = function handleFileChange(file) {
    _papaparse.default.parse(file, {
      header: true,
      dynamicTyping: true,
      complete: handleDataChange
    });
  };

  var makeColumns = function makeColumns(rawColumns) {
    return rawColumns.map(function (column) {
      return {
        Header: column,
        accessor: column
      };
    });
  };

  var handleDataChange = function handleDataChange(file) {
    setData(file.data);
    setColumns(makeColumns(file.meta.fields));
  };

  return _react.default.createElement("div", null, _react.default.createElement(_CsvInput.default, {
    handleFileChange: handleFileChange,
    data: data
  }), !loading && _react.default.createElement(_reactTable2.default, {
    data: data,
    columns: columns,
    defaultPageSize: 10,
    className: "-striped -highlight"
  }));
}

var _default = Interface;
exports.default = _default;
