// react-csv-viewer/dist/CsvInput.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireWildcard(require("react"));

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

function CsvInput(_ref) {
  var handleFileChange = _ref.handleFileChange;
  var csvRef = (0, _react.useRef)(null);

  var handleChange = function handleChange(e) {
    var file = csvRef.current.files[0];
    if (!file) return;
    handleFileChange(file);
  };

  return _react.default.createElement("form", {
    className: "csv-form"
  }, _react.default.createElement("div", {
    className: "csv-form-group"
  }, _react.default.createElement("input", {
    type: "file",
    className: "csv-form-control",
    id: "file",
    accept: ".csv",
    onChange: handleChange,
    ref: csvRef
  })));
}

var _default = CsvInput;
exports.default = _default;
